lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
+                        }
+                    }
                 }
             }
+            return false
+        }).map(pt => {
+            const itemName = pt.username || pt?.metadata?.length > 8 && this.getItemDefinitionById(pt.metadata[8]?.itemId)?.name || pt.name
+            let distance = distances[pt.id]
+            if (!distance) {
+                distance = this.position().distanceTo(pt.position)
+                distances[pt.id] = distance
+            }
+            const value = itemDataMap[itemName]
+            const sortValue = sortValueFunction(value, distance)
+            return new FindResult(pt, sortValue)
+        }).sort((a, b) => {
+            return a.value-b.value
         });
+
+        this.#logDebug(`Detected ${result.length} useful items to collect in range`)
+
+        result.length = Math.min(result.length, maxCount || result.length)
+
+        return result
+    }
+
+    /**
+     * Collects the item from the ground if it exists and is on the ground.
+     *
+     * @param {Entity} item
+     * @returns {Promise<boolean>} True if an item was collected
+     */
+    async collectItemOnGround(item) {
+        let result = false;
+        if (item && item.objectType === "Item" && item.onGround) {
+            const collectFunction = async (collector, collected) => {
+                if (collector.username === this.username()) {
+                    this.#logDebug(`I collected: ${collected.username || collected?.metadata?.length > 8 && this.getItemDefinitionById(collected?.metadata[8]?.itemId)?.name || collected.name}`);
+                    result = true
+                }
+            }
+
+            // Use collection events to track pickups
+            this.#bot.on('playerCollect', collectFunction);
+            try {
+                await this.approachEntity(item)
+            } catch (ex) {
+                console.warn(`Failed to approach item, error: ${ex}`)
+                // we failed to approach something on the ground... if we don't catch this
+                // we'll get stuck trying to pick up this item we can't pick up forever
+            }
+            this.#bot.off('playerCollect', collectFunction);
+        } else {
+            this.#log(`Unable to collect item: ${item && item.name || "null"}.  Not an 'Item' or not on the ground.`)
+        }
+        return result
     }
 
     /**
      * Collects all Items on the ground within a maximum distance from the Bot.
-     * @param {object} options={} Optional parameters
-     * @param {string} [options.itemName=undefined] Find and collect only Items with this name
-     * @param {boolean} [options.partialMatch=false] If itemName is defined, find Items whose names / displayNames contain itemName. (Ex. 'boots' may find any of 'iron_boots', 'golden_boots', etc.).
+     * @param {object} [options={}] Optional parameters
+     * @param {string} [options.itemNames=[]] Find and collect only Items with this name
```

Comment:
If I only want to find one item, does that mean I have to construct that array?
I think the code would be more readable if I had an option for just the item name... trying to think of the best way to structure that.

For context on my thinking - I am always asking myself the following "I am a player that has never look at RG bot. I want to pick an item off the ground. What would I expect the  signature to be".

In this case, I think there are three things a user may want to do
* Pick up all items on the ground
* Pick up a specific item on the ground
* Pick up a list of specific items on the ground
We should make sure all are straightforward.

I am very very much nitpicking a lot on this case.  I am just complaining about `[itemName]` vs `itemName`... but even this little tiny thing I think is important to think about. I will think more specifically about this case, I am not sold even on my own complaint.